daemons/ipa-slapi-plugins/ipa-pwd-extop/common.c

```diff
@@ -465,6 +465,23 @@ int ipapwd_entry_checks(Slapi_PBlock *pb, struct slapi_entry *e,
     *is_krb = slapi_entry_attr_has_syntax_value(e, SLAPI_ATTR_OBJECTCLASS, sval);
     slapi_value_free(&sval);
 
+    /* If entry has krbPrincipalAux object class but lacks krbPrincipalName and
+     * memberOf attributes consider this not a Kerberos principal object. In
+     * FreeIPA krbPrincipalAux allows to store krbPwdPolicyReference attribute
+     * which is added by a CoS plugin configuration based on a memberOf
+     * attribute value.
+     * Note that slapi_entry_attr_find() returns 0 if attr exists, -1 for absence
+     */
+    if (*is_krb) {
+        Slapi_Attr *attr_prname = NULL;
+        Slapi_Attr *attr_memberof = NULL;
+        int has_prname = slapi_entry_attr_find(e, "krbPrincipalName", &attr_prname);
+        int has_memberOf = slapi_entry_attr_find(e, "memberOf", &attr_memberof);
+        if ((has_prname == -1) && (has_memberOf == 0)) {
+            *is_krb = 0;
```

Comment:
Ok I think this is the cause of password history being simply not set therefore never matching.
As this line here:
  https://github.com/freeipa/freeipa/blob/0aeccc08524931b0e37773b4836734bc32eca7f4/daemons/ipa-slapi-plugins/ipa-pwd-extop/prepost.c#L1059
causes this code to bail out before ipapwd_setPasswordHistory() is called a few lines below.